include/hc_printf.hpp

```diff
@@ -286,6 +336,7 @@ static inline void processPrintfPackets(PrintfPacket* packets, const unsigned in
     formatString = std::regex_replace(formatString,doubleAmpersandPattern,"%");
     std::printf("%s",formatString.c_str());
   }
+  std::flush(std::cout);
 }
 
 static inline void processPrintfBuffer(PrintfPacket* gpuBuffer) {
```

Comment:
@scchan if i understand the logic correctly, `processPrintBuffer` would still be invoked with a non-`nullptr` `gpuBuffer`. would it be possible to further optimize this?

Reply:
If you refer to the following line, the host would first check whether the printf buffer contains any new data before proceeding to do further processing.  If kernels in the applicaton doesn't print anything the overhead should be minimal

https://github.com/RadeonOpenCompute/hcc/pull/498/files#diff-531a057446b7d96d716399f005b31ad8R349

Reply:
got it. with the revised logic it's more clear.